[FEAT] add OAuth key integration and additional URL-style params to --auth-key flags (ephemeral & preauthorized) (#399)
artis3n/ansible-role-tailscale: README.md

This role installs and configures [Tailscale][] on a Linux target.

Supported operating systems:

> 

Reviewer: Should be fine without the extra empty line, right?

A Tailscale Node Authorization auth key.

A Node Authorization key can be generated under your Tailscale account. The role supports two type of keys:

- Auth key (`tskey-auth-XXX-YYYYY`) <https://login.tailscale.com/admin/authkeys>
- OAuth key (`tskey-client-XXX-YYYY`) <https://login.tailscale.com/admin/settings/oauth>

> Note that reusable auth keys now expire 90 days after they are generated.

Reviewer: Let's modify to something like:

```suggestion
Note that auth keys expire up to a maximum of 90 days after they are generated. OAuth secrets do not expire unless revoked, and the generated OAuth access token expires after 1 hour.

For more information, see Tailscale's [OAuth clients](https://tailscale.com/kb/1215/oauth-clients) page, especially [Generating long-lived auth keys](https://tailscale.com/kb/1215/oauth-clients#generating-long-lived-auth-keys).
```

This value should be treated as a sensitive secret.

> #### tailscale_ephemeral

Reviewer: ```suggestion
### tailscale_oauth_ephemeral
```

> **Used only if `tailscale_authkey` is OAuth key**

Reviewer: ```suggestion
> [!NOTE]
> Used only when `tailscale_authkey` is an OAuth key.
```

Reviewer: Syntax from: https://github.com/orgs/community/discussions/16925

**Default**: `true`

> Register as an ephemeral node, if `true`.

Reviewer: ```suggestion
Register as an [ephemeral node](https://tailscale.com/kb/1111/ephemeral-nodes), if `true`.
```

> #### tailscale_preauthorized

Reviewer: ```suggestion
### tailscale_oauth_preauthorized
```

> **Used only if `tailscale_authkey` is OAuth key**

Reviewer: ```suggestion
> [!NOTE]
> Used only when `tailscale_authkey` is an OAuth key.
```

**Default**: `false`

> Skip manual device approval if `true`.

Reviewer: ```suggestion
Skip [manual device approval](https://tailscale.com/kb/1099/device-approval) if `true`.
```
